### ISSUE TITLE
the colin27_zlib.bnii sample has an uint8 Affine matrix

### ISSUE THREAD

**Issue by robertoostenveld** [fangq/jnifti#4]
```
>> jnii = loadjnifti('colin27_zlib.bnii')
jnii = 
  struct with fields:

    NIFTIHeader: [1×1 struct]
      NIFTIData: [181×217×181 uint8]
>> jnii.NIFTIHeader.Affine
ans =
  3×4 uint8 matrix
   1   0   0   0
   1   0   0   0
   1   0   0   0
```

I guess that the the `srow_x`, `srow_y` and `srow_z` arrays are used to make the Affine matrix. According to https://brainder.org/2012/09/23/the-nifti-file-format/ these arrays should be float (i.e., single precision).

Looking more critically at the correspondence between nifti and jnifti, I would actually say that there should not be a 3x4 Affine array, but that it should have the 1x4 `srow_x`, `srow_y` and `srow_z` arrays.

**Comment by fangq** [comment]
the matrix layout issue is now fixed in #2.

regarding the data format, bnii is based on the [BJData](https://github.com/NeuroJSON/bjdata/blob/master/Binary_JData_Specification.md) format that was extended from UBJSON (http://ubjson.org). It is not requrired, but UBJSON's spec recommends storing the data in the [smallest numeric type](https://github.com/ubjson/universal-binary-json/blob/master/spec12/value-types.html#L259) without losing precision. So, in this case, because the numbers are all 0s and 1s, `savebj.m` decides that the smallest data type is `uint8`. This behavior is consistent with other UBJSON writers/readers (such as [py-ubjson](https://github.com/NeuroJSON/pybj/blob/master/bjdata/encoder.py#L62-L89) and [json for modern C++](https://github.com/nlohmann/json/blob/develop/include/nlohmann/detail/output/binary_writer.hpp#L1436-L1515))

at this current version, JNIfTI does not require JNIfTI header fields must match NIfTI header type (since JSON is not strongly-typed). The conversion that reinforces that happens inside [`jnii2nii.m`](https://github.com/NeuroJSON/jnifty/blob/master/jnii2nii.m#L151-L153) if you are converting jnii back to nii.

let me know if this is a concern - we always have the possibility to change the spec, but I want to keep it as minimal as possible.

**Comment by fangq** [comment]
saw you had added type casting in this line https://github.com/robertoostenveld/fieldtrip/commit/a9855adb3700a286e7751aa1f14fcc232412641c#diff-6e27ee2a895fbaba1d39bf137a9e2d3168f79ce533c75ce1048e3180d8284716R641

assume this issue can be closed.
